File: DesignPatternsInAutomatedTesting-Series/PatternsInAutomation.Tests/Advanced/Strategy/Services/VatTaxCalculationService.cs
using PatternsInAutomation.Tests.Advanced.Strategy.Enums;

namespace PatternsInAutomation.Tests.Advanced.Strategy
{
    public class VatTaxCalculationService
    {
        public decimal Calculate(decimal price, Countries country)
        {
            decimal taxPrice = default(decimal);
            // Call Real Web Service to determine the VAT Tax.
            switch (country)
            {
                case Countries.Bulgaria:
                    taxPrice = CalculateTaxPriceInternal(price, 20);
                    break;
                case Countries.Germany:
                    taxPrice = CalculateTaxPriceInternal(price, 19);
                    break;
                case Countries.Austria:
                    taxPrice = CalculateTaxPriceInternal(price, 20);
                    break;
                case Countries.France:
                    taxPrice = CalculateTaxPriceInternal(price, 23);
                    break;
                default:
                    taxPrice = 0;
                    break;
            }

            return taxPrice;
        }

        private static decimal CalculateTaxPriceInternal(decimal price, double percent)
        {
            decimal taxPrice = price / (decimal)percent;
            return taxPrice;
        }
    }
}

File: DesignPatternsInAutomatedTestingJava-Series/src/test/java/decorator/services/VatTaxCalculationService.java
package decorator.services;

import decorator.enums.Country;

public class VatTaxCalculationService {
    double taxValue;
    public double calculate(double price, Country country) {
        switch(country) {
            case BULGARIA:
            case UNITED_KINGDOM:
            case GERMANY:
            case AUSTRIA:
            case FRANCE:
                taxValue = calculateVATInternal(price, 20);
                break;
            default:
                taxValue = 0;
                break;
        }

        return taxValue;
    }

    private static double calculateVATInternal(double price, double percent) {
        var taxValue = price*percent/100;
        return taxValue;
    }
}
